### DISCUSSION TITLE
Option to switch between subject types

### DISCUSSION THREAD

**Discussion by benefron** [bonsai-rx/bonsai#1187] [Q&A]
Hi,
Is there a quick way to change subject type (Behavior to publish/async, for example)?
Also, is there an option to quickly change all instances of a subject (Initial, multicast, etc.) to a different name?

**Comment by bruno-f-cruz** [comment]
Yes for both.
To replace a subject type, select the node that defines the subject. In the toolbox, find the subject type that you want to replace it by, right-click and select `Replace`.

To change the name of all instances of a given subject: select a subject in your workflow or toolbox, and click `F2` (or right click the toolbox entry and select `Rename`). It will allow you to edit its name and propagate it to all instances.

Hope it helps,
Bruno
